### PR TITLE
xfail Python tests dependent on invalid Polygons

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Appears the `gs` fixture in the `conftest.py` contains invalid polygon (the holes are beyond shell), and geos 3.13 now does not consider the same invalid polygon equivalent https://github.com/libgeos/geos/issues/1177.

The `gs` fixture and the `polys` fixture (which is hard-coded) need to both be updated, but I wasn't able to get a quick-working fix (I'm not familiar with this area). This PR just xfails the effected tests.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by vyasr — APPROVED** [review]
Approving to unblock the 24.10 release. It would be good to get confirmation from a cuspatial developer more familiar with these tests to determine if these tests are important and indicative of some real issue that we need to address, but I don't think it's worth having cuspatial hold up the release so if needed we can release a hotfix.

File: python/cuspatial/cuspatial/tests/conftest.py
Comment (harrism) [inline python/cuspatial/cuspatial/tests/conftest.py:33]: These are multipolygons, not polygons, so I don't think they are invalid!

**Pull request merged**
